File: backend/app/schemas/reservation.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field, ConfigDict
from app.schemas.flight import FlightResponse


class PassengerCreate(BaseModel):
    first_name: str = Field(..., min_length=1, max_length=100)
    last_name: str = Field(..., min_length=1, max_length=100)
    gender: Optional[str] = Field("M", max_length=10)
    seat_preference: Optional[str] = Field(None, max_length=10)


class PassengerResponse(BaseModel):
    id: str
    first_name: str
    last_name: str
    gender: Optional[str] = None
    seat_number: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class ReservationCreate(BaseModel):
    flight_id: str
    passengers: List[PassengerCreate] = Field(..., min_length=1, description="List of passengers for booking")


class ReservationResponse(BaseModel):
    id: str
    pnr: str
    user_id: str
    flight: FlightResponse
    total_amount: float
    status: str
    passengers: List[PassengerResponse]
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class PaginatedReservationResponse(BaseModel):
    items: List[ReservationResponse]
    total: int
    page: int
    limit: int
    total_pages: int

File: frontend/src/types/reservation.ts
import { Flight } from './flight';

export interface Passenger {
  id: string;
  first_name: string;
  last_name: string;
  gender?: string;
  seat_number?: string;
}

export interface PassengerCreatePayload {
  first_name: string;
  last_name: string;
  gender?: string;
  seat_preference?: string;
}

export interface Reservation {
  id: string;
  pnr: string;
  user_id: string;
  flight: Flight;
  total_amount: number;
  status: 'CONFIRMED' | 'CANCELLED';
  passengers: Passenger[];
  created_at: string;
  updated_at: string;
}

export interface ReservationCreatePayload {
  flight_id: string;
  passengers: PassengerCreatePayload[];
}

export interface PaginatedReservationResponse {
  items: Reservation[];
  total: number;
  page: number;
  limit: number;
  total_pages: number;
}
